aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -525,6 +528,155 @@ void cpu_hflip_vec(at::TensorIterator& iter) {
   iter.cast_outputs();
 }
 
+void cpu_hflip_channels_last_uint8_ch3(at::TensorIterator& iter) {
+
+  auto loop2d = [&](char** base, const int64_t *strides, int64_t size0, int64_t size1) {
+
+    // Here ntensors is defined for output and 1 input. But tensor iterator has defined output, input
+    // and restrided_input (see aten/src/ATen/native/TensorTransformations.cpp#L64-L66) but we use only
+    // output and input.
+    static constexpr int ntensors = 2;
+    const int64_t *outer_strides = &strides[3];
+    const int64_t stride = strides[0];
+
+    TORCH_INTERNAL_ASSERT(strides[0] == strides[1]);
+    TORCH_INTERNAL_ASSERT(stride == 1);
+    TORCH_INTERNAL_ASSERT(outer_strides[0] == -3);
+    TORCH_INTERNAL_ASSERT(outer_strides[1] == 3);
+
+    char* C10_RESTRICT data[ntensors] = {base[0], base[1]};
+    const int64_t size = size0 * size1;
+
+    int64_t i = 0;
+
+#ifdef CPU_CAPABILITY_AVX2
+
+    if (size0 == 3) {
```

Comment:
You should add a test for this.

Reply:
There is a check here : https://github.com/pytorch/pytorch/pull/91806/files#diff-54b494a4dd0af2160d716378bd5a40e1e4a98c94414901d85adbb6a9ae6dbed2R709
and this `if` here is for `OMP_NUM_THREADS>1` when it can split size0 into smaller chunks.

Reply:
I mean a test in CI calling flip with N3HW input.